### ISSUE TITLE
[Bug]: Holding "Shift" in FTB Quest brings up a search menu

### ISSUE THREAD

**Issue by Xiarno** [FTBTeam/FTB-Modpack-Issues#1696]
### Modpack

FTB Stone Block 3

### Modpack version

1.1.0

### Describe the bug

When holding shift in FTB Quest, the search tab is brought up.
While I don't think it's a "bug" in itself, the problem comes up when looking at the item from some quest. A lot of those items have "press shift for more info" or something like that. So it makes it really annoying to have that each time.

### Steps to reproduce

1. Go to the FTB Quest menu.
2. Hold shift at any point. (More importantly, while looking at an item for a quest.)
3. See the errors

### Expected behaviour

The "bring the search tab" key should be moved to something else.

**Comment by Sunekaer** [comment]
What error are you seeing? All it's showing me if you double-click shift is the search menu

**Comment by Xiarno** [comment]
> What error are you seeing? All it's showing me if you double-click shift is the search menu

Then I guess this is the error. The search menu is brought up just by holding shift. No need to double-click it. As I said, while I don't think it's a bug in itself, due to the quests showings loads of items where you hold shift to see more information, each time you try it brings up the search menu which prevents actually seeing the information of the quest item.

**Comment by desht** [comment]
Holding Shift should not bring up any search menu (and doesn't for me either). It seems like for some reason your keyboard is sending multiple Shift key-pressed events when you hold down Shift, which should not be the case.  What OS are you using (and have you noticed any problems with the shift key in other apps) ?

Having said, ignoring the double-tap-shift while the view quest panel is popped up would be an easy addition and would get round this problem you're seeing. And a keybind like Ctrl-F would probably make more sense as a search function in any case...

**Comment by Xiarno** [comment]
> Holding Shift should not bring up any search menu (and doesn't for me either). It seems like for some reason your keyboard is sending multiple Shift key-pressed events when you hold down Shift, which should not be the case. What OS are you using (and have you noticed any problems with the shift key in other apps) ?
> 
> Having said that, ignoring the double-tap-shift while the view quest panel is popped up would be an easy addition and would get round this problem you're seeing. And a keybind like Ctrl-F would probably make more sense as a search function in any case...

I'm using Windows 10 and no, I've not seen any issue with the shift key on any other app.

Yes, a Ctrl-F would make more sense.

**Comment by desht** [comment]
Can you try this web application: https://codepen.io/QDeltaE/full/PjXWZr

Click the keyboard, then hold shift down.  You should just `>Keydown ↓ Key: "Shift" KeyCode: 16` or similar, not repeated, and then `>Keyup   ↑ Key: "Shift" KeyCode: 16` when you release the Shift key.  If you see repeated Keydown/Keyup messages for the Shift key, then you have some problem with your keyboard auto-repeat.

**Comment by Xiarno** [comment]
> Can you try this web application: https://codepen.io/QDeltaE/full/PjXWZr
> 
> Click the keyboard, then hold shift down. You should just see `>Keydown ↓ Key: "Shift" KeyCode: 16` or similar, not repeated, and then `>Keyup ↑ Key: "Shift" KeyCode: 16` when you release the Shift key. If you see repeated Keydown/Keyup messages for the Shift key, then you have some problem with your keyboard auto-repeat.

Like this?

https://user-images.githubusercontent.com/24660485/201103208-bb21b1cd-09f1-4313-8782-9539a5332260.mp4

**Comment by Xiarno** [comment]
Nope. As far as I know, it's always been like that on all my computers and keyboard. As well as every letter. If I just hold anything down, it starts spamming it / holding it.

**Comment by desht** [comment]
Most keys should behave that way normally (unless an app specifically disables key repeat), but not modifier keys (shift/ctrl/alt).

**Comment by Xiarno** [comment]
> Most keys should behave that way normally (unless an app specifically disables key repeat), but not modifier keys (shift/ctrl/alt).

Idk they've always done that with me.

https://user-images.githubusercontent.com/24660485/201341136-5a51bbae-1988-41b4-a356-b359b50e2567.mp4

**Comment by DalekZED** [comment]
I also have this issue. Was there no resolution? I have this issue on desktop with Corsair k55, and two other keyboards but not with my laptop. Have tried it on Win 10, 11, and Fedora. Seems to be a hardware thing.

**Comment by Shadowphinex** [comment]
hi i know this is old but i had this  problem again today so digging a little if you go to the section shown and turn on  just the filter keys setting it seams to fix it.
![Screenshot 2024-11-05 120733907](https://github.com/user-attachments/assets/bc8d1e8c-0edc-400d-9b52-003a1c3ea3bb)

**Comment by romandetect** [comment]
same question.i need help,too.

**Comment by desht** [comment]
Just to note, this is a non-issue in modern versions of the mod, since the keybind for search is now Ctrl+F.
